docc.go

```diff
@@ -2,168 +2,226 @@ package docc
-	if ext == ".doc" {
-		xfp, err := docToX(docxPath)
-		if err != nil {
-			return nil, fmt.Errorf("fail or not support .doc file: %w", err)
-		}
-		r.docxPath = xfp
-		r.fromDoc = true
-	} else if ext != ".docx" {
+	if ext != ".docx" {
 		return nil, ErrNotSupportFormat
 	}
 
-	a, err := zip.OpenReader(r.docxPath)
+	zipReadCloser, err := zip.OpenReader(reader.docxPath)
 	if err != nil {
 		return nil, err
 	}
-	r.docx = a
 
-	f, err := a.Open("word/document.xml")
-	if err != nil {
-		return nil, err
+	var fileReaders []FileReader
+	for _, file := range zipReadCloser.File {
+		if file.Name == "word/document.xml" ||
+			strings.Contains(file.Name, "header") ||
+			strings.Contains(file.Name, "footer") ||
+			strings.Contains(file.Name, "footnotes") {
+			openedFile, err := zipReadCloser.Open(file.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			fileReaders = append(fileReaders, FileReader{
+				fileName: file.Name,
+				xml:      openedFile,
+				decoder:  xml.NewDecoder(openedFile),
+			})
+		}
 	}
-	r.xml = f
-	r.dec = xml.NewDecoder(f)
 
-	return r, nil
+	reader.fileReaders = fileReaders
+	reader.docx = zipReadCloser
+
+	return reader, nil
 }
 
-// Read reads the .docx file by a paragraph.
-// When no paragraphs are remained to read, io.EOF error is returned.
-func (r *Reader) Read() (string, error) {
-	err := seekNextTag(r.dec, "p")
+func (r *Reader) read(decoder *xml.Decoder) (string, error) {
+	err := seekNextTag(decoder, xmlParagraph)
 	if err != nil {
 		return "", err
 	}
-	p, err := seekParagraph(r.dec)
+	paragraph, err := seekParagraph(decoder)
 	if err != nil {
 		return "", err
 	}
-	return p, nil
+	return paragraph, nil
 }
 
-// ReadAll reads the whole .docx file.
-func (r *Reader) ReadAll() ([]string, error) {
-	ps := []string{}
+func (r *Reader) readSingleFile(decoder *xml.Decoder) (string, error) {
+	var content strings.Builder
 	for {
-		p, err := r.Read()
+		paragraph, err := r.read(decoder)
 		if err == io.EOF {
-			return ps, nil
+			return content.String(), nil
 		} else if err != nil {
-			return nil, err
+			return "", err
+		}
+		content.WriteString(paragraph)
+		content.WriteString(" ")
+	}
+}
+
+func (r *Reader) ReadAllFiles() (headerValue, contentValue, footerValue string, err error) {
+	var header, footer, content strings.Builder
+	for _, fileReader := range r.fileReaders {
+		fileContent, err := r.readSingleFile(fileReader.decoder)
+		if err != nil {
+			return "", "", "", err
+		}
+
+		if strings.Contains(fileReader.fileName, "header") {
+			header.WriteString(fileContent)
+		} else if strings.Contains(fileReader.fileName, "footer") {
+			footer.WriteString(fileContent)
+		} else {
+			content.WriteString(fileContent)
 		}
-		ps = append(ps, p)
 	}
+
+	return header.String(), content.String(), footer.String(), nil
+}
+
+func (r *Reader) CloseReaderFromBytes() error {
+	for _, fileReader := range r.fileReaders {
+		fileReader.xml.Close()
+	}
+	if r.fromDoc {
+		os.Remove(r.docxPath)
+	}
+	return nil
```

Comment:
It seems error-prone to have two separate `Close*` methods, since it depends on the client to call the correct one. Perhaps we could have just one `Close() method` with some conditional logic to handle either the bytes-based or file-based reader. Example:
```
if r.docx != nil {
  r.docx.Close()
}
```